File: CP2K_functions_Py/Functions_pwd/Py_Functions_files/LocateStationaryPoints.py
import numpy as np
    
def LocateStationaryPoints(f = None): 
    dydx = np.zeros((f.shape[1-1],f.shape[2-1]))
    for i in np.arange(1,f.shape[2-1]+1).reshape(-1):
        #     dydx(:,i) = gradient(smooth(f(:,i))); # smoothed function misses
#     sharp minima...
        dydx[:,i] = gradient(f(:,i))
        count = []
        for j in np.arange(1,len(dydx) - 1+1).reshape(-1):
            if dydx(j) < np.logical_and(0,dydx(j + 1)) > 0:
                count = np.array([count,j])
        MinimaIndx[i] = count
    
    return MinimaIndx
    return MinimaIndx

File: CP2k_functions/Main_pwd/Charge_Analysis_Rashid/Charge_density/LocateStationaryPoints.m
function [MinimaIndx] = LocateStationaryPoints(f)

dydx = zeros(size(f,1), size(f,2));
for i=1:size(f,2)
    %     dydx(:,i) = gradient(smooth(f(:,i))); % smoothed function misses
    %     sharp minima... 
    dydx(:,i) = gradient(f(:,i)); % non-smoothed function picks up noise
    
    count = [];
    for j = 1:length(dydx)-1
        if dydx(j) < 0 & dydx(j+1) > 0
            count = [count j];
        end
    end
    
    MinimaIndx{i} = count;
end

return

